library/ImageControl.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+import json
+
+from library.common import *
+
+class ImageControl(object):
+    def __init__(self, urlhandler):
+        self.urlhandler = urlhandler
+
+    def check_for_existence(self, img_type, desc, arch):
+        url = "/api/v1/image?show_all=true"
+        allowed_types = ["Kernel", "Initrd", "bootloader"]
+        if img_type not in allowed_types:
+            raise ProvisionerError("Error: Image type is '{}'; must be one of {}".format(
+                                img_type, allowed_types))
+
+        images = self.urlhandler.get(url)
+
+        for image in images:
+            if (image['description'] == desc and
+                image['type'] == img_type and
+                image['arch'] == arch):
+                return image
```

Comment:
check_for_existence returns the image itself, which is incongruent. This is also used to get the image contents, so it should really be called get_image, and be used in its boolean context to check for existence:
```
if not get_image(...):
    raise Error(no image)
```

Reply:
Renamed get_image, but now returns either the image (so dictionary basically) or Null to keep it consistent.